From 9ef2a5515196a9344305c74ddd08f9e698cfae9e Mon Sep 17 00:00:00 2001
From: Steve deRosier <Steve.Derosier@lairdtech.com>
Date: Fri, 20 Sep 2013 17:30:30 -0700
Subject: [PATCH] wb45n-dt: Add gpio_keys on IRQ btn

This adds the IRQ button as a gpio_keys driver and assignees the sysreq key.
Pressing this button will wake the wb45n from sleep.
---
 arch/arm/boot/dts/at91sam9x5cm.dtsi |   13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/boot/dts/at91sam9x5cm.dtsi b/arch/arm/boot/dts/at91sam9x5cm.dtsi
index 623d19c..e2d85e2 100644
--- a/arch/arm/boot/dts/at91sam9x5cm.dtsi
+++ b/arch/arm/boot/dts/at91sam9x5cm.dtsi
@@ -100,4 +100,17 @@
 			gpios = <&pioC 14 0>;
 		};
 	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		irqbtn@pb18 {
+			label = "IRQBTN";
+			linux,code = <99>;
+			gpios = <&pioB 18 1>;
+			gpio-key,wakeup = <1>;
+		};
+	};
+
 };
-- 
1.7.9.5

